spring-boot-project/spring-boot-tools/spring-boot-maven-plugin/src/main/java/org/springframework/boot/maven/RepackageMojo.java

```diff
@@ -222,9 +222,12 @@ private void repackage() throws MojoExecutionException {
 	}
 
 	private FileTime parseOutputTimestamp() {
-		// Maven ignores a single-character timestamp as it is "useful to override a full
-		// value during pom inheritance"
-		if (this.outputTimestamp == null || this.outputTimestamp.length() < 2) {
+		// Maven ignores a single-character non-digit timestamp as it is
```

Comment:
It feels to me that the right fix would be to declare a dependency on the maven-archiver artifact and call `MavenArchiver#parseBuildOutputTimestamp(String outputTimestamp)` to get the exact same behavior (and then parse the result into a FileTime object). This is still incomplete and may break in other ways.

See https://github.com/nielsbasjes/maven-archiver/blob/master/src/main/java/org/apache/maven/archiver/MavenArchiver.java#L768C37-L801 for the actual implementation.

Reply:
Thanks. We'll polish the PR when we merge it and then use the same code as your linked Mojo.